File: EndlessWorld/Assets/Collision HIT/Scripts JS/Dang/Add/TriggerControl.js
#pragma strict

var MainObject:GameObject;
var PlayAnimation:boolean=false;
var AddForces:Vector3;
var AddRotation:Vector3;


private var DontRepeat:boolean = false;


function OnTriggerEnter(other: Collider){ 
if (gameObject.GetComponent.<TriggerControl> ().enabled == true) {
if(other.gameObject.tag=="Camera"){
if(!DontRepeat){ DontRepeat = true;
if(PlayAnimation){
MainObject.GetComponent.<Animation>().Play();
}else{
MainObject.GetComponent.<Rigidbody>().AddForce(AddForces*1);
MainObject.GetComponent.<Rigidbody>().angularVelocity = AddRotation*1;
}
}
}
}
}

File: EndlessWorld/Assets/Collision HIT/Scripts C#/Dang/Add/TriggerControl.cs
using UnityEngine;
using System.Collections;

public class TriggerControl : MonoBehaviour {

	public GameObject MainObject;
	public bool PlayAnimation = false;
	public Vector3 AddForces;
	public Vector3 AddRotation;

	private bool DontRepeat = false;

	void OnTriggerEnter(Collider other){ 
		if (gameObject.GetComponent<TriggerControl> ().enabled == true) {
			if (other.gameObject.tag == "Camera") {
				if (!DontRepeat) {
					DontRepeat = true;
					if (PlayAnimation) {
						MainObject.GetComponent<Animation> ().Play ();
					} else {
						MainObject.GetComponent<Rigidbody> ().AddForce (AddForces * 1);
						MainObject.GetComponent<Rigidbody> ().angularVelocity = AddRotation * 1;
					}
				}
			}
		}
	}

	void Update(){


	}


}
